{% from 'layout/carousel.html' import carousel_navigation with context %}
{% extends "layout/layout.html" %}
{% set add_sidebar = true %}

{% block article_content %}
  {{ subheading('Goals and Purpose') }}
    <div>
      <p>
        The main goal of the wet lab team is to test and validate, in vitro, the AI-generated plasmid parts the dry lab 
        team creates. Validating each individual plasmid part allows us to determine with more granularity which 
        specific AI-plasmid components are functional.
      </p>
      <p>
        To achieve this goal, we utilize a pipeline in which we assemble the AI-parts into a known plasmid backbone 
        using Golden Gate Assembly (GGA – a high-throughput transgene insertion method), and test the viability of the 
        AI-parts using bacterial growth assays. Finally, we sequence the new working plasmid parts to confirm 
        successful propagation of the AI-parts, such as the origin of vegetative replication (oriV) and the 
        toxin-antitoxin (TA) system. These developments provide a proof-of-concept for the tractability of synthesizing 
        plasmids de novo using dry lab’s machine learning framework, which sets the path forward to ultimately create 
        fine-tuned plasmids to tackle antibiotic resistance.
      </p>
      <p>
        The wet lab pipeline for validating various AI-plasmid parts, in particular the oriV and antibiotic 
        resistance genes (ARGs), is described in detail in the following sections. Thus far, the wet lab team has 
        synthesized and tested novel oriVs generated by dry lab, of which samples 741, 9371, 5276, 5727, and 4089 
        have been determined as functional. 
      </p>
    </div>
  {{ subheading('Design') }}
  {{ subsubheading('1. Backbone and Controls') }}
    <div>
      <p>
        The main AI-plasmid part the wet lab team tested was the oriV. In general, the AI-generated oriVs consisted of 
        various control elements (such as RNAI-, RNAII-, rep- and rop-coding regions) and non-coding regions. As such, 
        to test this part, the wet lab team cloned the dry lab oriVs into suitable backbones of which have similar 
        control elements for their oriVs. This ensured compatibility between the dry lab oriV and backbone.
      </p>
      <p>
        A backbone conferring resistance to a specific antibiotic serves as an effective positive and negative control 
        for growth assays that are essential for ascertaining dry lab oriV functionality. Such a backbone eliminates 
        confounding variables that may skew the interpretation of experimental results by observing E. coli growth as 
        well as number of colonies to determine if the backbone was successfully annealed to the ori, and successfully 
        uptaken into the E. coli.  
      </p>
      <p>
        Initially, the plasmid pSB1C3 was used as a backbone. pSB1C3 is a common plasmid backbone from E. coli which 
        confers chloramphenicol resistance. Thus, pSB1C3 was to be used as a positive control for the experiments, 
        since it was predicted that it would successfully grow on the plates. pSB1C3 was miniprepped and underwent PCR 
        to increase the amount of this plasmid, transformed  into E. coli, which was then plated. However, no growth 
        was observed from the transformed E. coli on the chloramphenicol plates. 
      </p>
      <p>
        After we repeated this experiment a number of times, the team decided to have the plasmid sequenced to check if 
        the structure was correct. Sequencing results revealed that the backbone contained additional BsaI sites than 
        initially described, meaning that it was being cut in additional locations, meaning the ori and the backbone 
        would fail to be annealed together. 
      </p>
      <p>
        This meant that afterward, a different positive control had to be used for the experiments. Interlab_NegC was 
        used, however, there it exhibited a similar problem as pSB1C3. After no growth was observed again, 
        Interlab_NegC was taken for sequencing and found to have an extra BsaI site, which caused the backbone to be 
        cleaved during GGA as our GGA workflow utilized the exact same restriction sites. 
      </p>
      <p>
        Finally, we found that eforRed/eforCP Biobrick K592012, an ampicillin (herein, Amp) backbone was successful as 
        a positive control backbone, indicated by numerous colonies on Amp. agar plates. This led us to use this 
        backbone for future experiments. eforRed/eforCP Biobrick K592012 had a length of 2807 bp, and contained a 861 
        bp long bla gene, coding for beta-lactamase and thus conferring Amp antibiotic resistance.
      </p>
      <p>
        <u>Positive Control</u><br>
        The positive control served to validate the wet lab team’s experimental approach of cloning dry lab oriVs into 
        functional backbones to test their functionality. Here, this control is carried out by cloning a known working 
        oriV (pJump27) into eforRed/eforCP Biobrick K592012 using GGA. This would theoretically create a construct that 
        confers Amp resistance provided that the cloned oriV is functional.
      </p>
      <p>
        As such, this construct is then transformed into competent E. coli K12 thawed on ice. The culture was then 
        plated on Amp plates, incubated at 37°C overnight, and then observed for growth of colonies the following 
        morning (which would indicat successful oriV functionality).
      </p>
      <p>
        Once we verified that this assembled plasmid was functional by plating transformed bacteria, we began our 
        assemblies using the Dry Lab-generated sequences.
      </p>
      <p>
        <u>Negative Control</u><br>
        The negative controls are made by transforming competent E. coli cells with a functional plasmid lacking an 
        oriV sequence. The negative control contains only a linear plasmid backbone without an ori, since it has not 
        been cloned and annealed to anything. When assembled and plated onto Amp plates, these cells will theoretically 
        die as they have no oriV to propagate any Amp-resistance cassettes. 
      </p>
      <p>
        The negative control is used to ensure the assembly’s procedure was correct, as E. coli without the ori to 
        replicate the plasmid would not survive on the antibiotic-plated media. For our GGA using the Interlab plasmid, 
        Interlab_NegC with water was used as a negative control. 
      </p>
    </div>
  {{ subsubheading('2. Cell Preparation') }}
    <div>
      <p>
        <u>Competent Cells</u><br>
        In our experiment, we used E coli DH5α cells due to their high efficiency in transformation and stable 
        amplification (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4562628/). These cells were stored at -80°C and 
        subsequently thawed on ice before use. To initiate the transformation, we added the assembled  plasmid, 
        containing the OriV drylab generated sequence with the eforCP backbone, to the thawed cells and incubated the 
        mixture. We then performed a heat shock by incubating the cells at 42°C for exactly 30 seconds to facilitate 
        the uptake of the plasmid DNA. Following heat shock, we added SOC medium to the cells and then incubated the 
        mixture at 37°C to ensure sufficient recovery. Finally, we plated the transformed cells on agar plates 
        containing AMP to select for successful transformants.
      </p>
    </div>
  {{ subsubheading('3. Assembly') }}
      <div>
      <p>
        Golden Gate Assembly (GGA) is a one-pot molecular cloning technique that allows for the combination of multiple 
        inserts into a backbone without the need for multiple restriction enzymes (Figure 1). We used GGA to assemble 
        all our plasmids for testing of oriV sequences and antibiotic resistance genes (ARGs).
      </p>
      <div>
        {{ img(new_cdn("https://static.igem.wiki/teams/5230/wetlab/gga-figure-1.webp"), 
                        alt="Golden Gate Assembly method diagarm") }}
      <p>
        <b>Figure 1.</b> Golden Gate Assembly uses type IIS restriction enzymes, which have separate recognition 
        (yellow) and cut sites (green, purple, light pink). This allows for the creation of many possible unique 
        overhangs without the need for multiple restriction enzymes. GGA can then proceed in a one-pot reaction, 
        allowing for a backbone and multiple inserts to be combined simultaneously. Figure created with BioRender.com.
      </p>
      </div>
      <p>
        We developed a molecular cloning (MoClo) system for our oriV and ARG assemblies (Figure 2). Inserts for both 
        experiments were assembled into backbones with a selection marker, the ARG backbone also containing an oriV 
        sequence (Figures 4-5). Dry lab generated oriV sequences were either well aligned to the oriV control element 
        RNAII or the noncoding and Rop sequence, so 5 different assemblies were used to test their functionality 
        (Figure 3).
      </p>
      <div>
        {{ img(new_cdn("https://static.igem.wiki/teams/5230/wetlab/moclo-system-figure-2.webp"), 
                        alt="iGEM Toronto MoClo system overhangs and their assigned letters") }}
      <p>
        <b>Figure 2.</b> OriV and ARG MoClo system overhangs and their assigned letters. Overhang locations in 
        assemblies are indicated in Figures 3 and 4. Figure created with BioRender.com.
      </p>
      </div>
      <div>
        {{ img(new_cdn("https://static.igem.wiki/teams/5230/wetlab/type-b-oriv-assembly-figure-3.webp"), 
                        alt="Type B oriV and sequences used to test generated oriV alignments") }}
      <p>
        <b>Figure 3.</b> Type B oriV and sequences used to test generated oriV alignments. Dry lab has been generating 
        type B oriV sequences, which contain sequences for RNAI, RNAII, Rop, and a noncoding region. Generated oriV 
        sequences have been aligning best to either RNAII or the noncoding region and Rop sequence, so we made 
        assemblies to test these generated sequences by themselves and with the sequence that they are best aligned to 
        from a wild type (WT) oriV sequence. Figure created with BioRender.com.
      </p>
      </div>
      <div>
        {{ img(new_cdn("https://static.igem.wiki/teams/5230/wetlab/oriv-assembly-maps-figure-4.webp"), 
                        alt="Assemblies used to test generated oriV functionality") }}
      <p>
        <b>Figure 4.</b> The assemblies used to test generated oriV functionality, letters indicating the locations of 
        specific MoClo overhangs. Each generated oriV sequence was tested by itself and with a portion of a WT oriV 
        containing the region of the oriV sequence that the generated sequence wasn’t best aligned to. The entire WT 
        oriV was also used as a control. Amp (Bla) was chosen as the selection marker. Figure created with 
        BioRender.com.
      </p>
      </div>
      <div>
        {{ img(new_cdn("https://static.igem.wiki/teams/5230/wetlab/arg-assembly-map-figure-5.webp"), 
                        alt="Assembly used to test generated ARG functionalit") }}
      <p>
        <b>Figure 5.</b> The assembly used to test generated ARG functionality, letters indicating the locations of 
        specific MoClo overhangs. A selection marker was included to allow for selection if the generated ARG was 
        nonfunctional. Figure created with BioRender.com.
      </p>
      </div>
    </div>
  {{ subsubheading('4. Transformation') }}
    <div>
      <p>
          After finishing the assemblies, we transformed our backbone eforRed/eforCP Biobrick K592012 with different 
          dry lab generated oriV sequences. We use competent E. coli DH5a cells to transform the plasmids in. We add 
          2 uL (1 pg – 100 ng) of our vectors to thawed competent cells. We then incubate on ice and heat shock the 
          cells, which is followed by cell recovery using SOC media for 1 hour. Afterwards, we plate the liquid cell 
          culture on Agar plates with corresponding antibiotics. 
      </p>
      <p>
          To test the function of the antibiotic resistant genes, we will also transform the assembled plasmids into 
          competent E. coli DH5a cells. We will be using double antibiotic plates, which includes CmR, our antibiotic 
          selection marker, and Carbenicillin. Carbenicillin is a beta-lactam ring containing an antibiotic that our 
          antibiotic resistant genes defend against. 
      </p>
    </div>
  {{ subheading('Experiments') }}
  {{ subsubheading('Preliminary experiments (Controls)') }}
    <div>
        <p>
        We began with validating our positive control experiment using the pSB1C3 device and the oriV from the 
        pJUMP27-1A plasmid from the 2024 iGEM Distribution Kit Plate. However, this vector was found to have unreported 
        BsaI cut sites. Therefore, we switched to the Interlab_NegC device, but that was also discovered to also have 
        unreported BsaI cut sites. Finally, we used the eforRed_eforCP K592012 device and obtained visible colonies and 
        a valid sequencing result, thus verifying our positive control.
        </p>
    </div>
  {{ subsubheading('Main experiments (Validation)') }}
    <div>
        <p>
        Next, we began assembling the AI-generated sequences generated by our dry lab team into the eforRed/eforCP 
        K592012 device (BBa_J435099). The first batch of sequences generated by dry lab were not found to be functional 
        as the transformed bacteria did not grow into visible colonies. 6 generated oriVs from the second batch of 
        sequences generated visible colonies, and 5 of them were confirmed to be functional via whole plasmid 
        sequencing. Testing for the third batch of sequences, which includes sequences encoding for antibiotic 
        resistance genes and oriVs, is currently in progress. 
        </p>
    </div>
  {{ subsubheading('Calendar') }}
    <div x-data="{ galleryIndex: 0 }">
        <div x-show="galleryIndex === 0" class="w-full p-8">
            Page A
        </div>
        <div x-show="galleryIndex === 1" class="w-full p-8 break-words">
            B Side
        </div>
        <div x-show="galleryIndex === 2" class="w-full p-8 break-words">
            C Section
        </div>
        <div x-show="galleryIndex === 2" class="w-full p-8 break-words">
            D Section
        </div>
        <div x-show="galleryIndex === 2" class="w-full p-8 break-words">
            E Section
        </div>
       {{ carousel_navigation("galleryIndex", 3) }}
    </div>
  {{ subheading('Results') }}
    <div>
      <p>
        <u>Controls</u><br>
        We first began by assembling and validating our positive and negative controls. After switching over to the 
        eforRed_eforCP vector, we amplified the positive control pJUMP27 insert and verified via agarose gel 
        electrophoresis its correct size of 1.6 kb before purifying, transforming, and plating. 
      </p>
      <p>
        <u>Batch 1</u>
      </p>
      <p>
        Unfortunately, numerous attempts at transforming the first batch of AI-generated sequences did not produce 
        viable colonies on the plates. Since our positive control produced colonies, we concluded that this first batch 
        was nonfunctional. 
      </p>
      <p>
        The wet lab pipeline for validating various AI-plasmid parts, in particular the oriV and antibiotic 
        resistance genes (ARGs), is described in detail in the following sections. Thus far, the wet lab team has 
        synthesized and tested novel oriVs generated by dry lab, of which samples 741, 9371, 5276, 5727, and 4089 
        have been determined as functional. 
      </p>
      <p>
        <u>Batch 2</u>
      </p>
      <p>
        After completing the required workflow to prepare the inserts, we introduced our second batch of plasmids into 
        E. coli for uptake. This batch included eight new AI-generated oriV sequences, six of which produced viable 
        colonies on the plates. This result aligned with our expectations, as Batch 2 showed improved alignment with 
        the oriV control element RNAII. 
      </p>
      <p>
        We also tested the wild-type oriV which the sequences aligned to, pColE, and saw viable colonies, which 
        provides a reliable point of comparison for the AI-generated oriVs. The positive control, pJUMP27, also 
        produced viable colonies, indicating that oriVs are indeed modular. Upon cloning this batch, we found that 
        sequences with the IDs 741, 4089, 5276, 5727, 6769, and 9371 produced visible colonies. Upon sequencing, we 
        confirmed that the sequences with the ID 741, 4089, 5276, 5727, 9371, and the wild-type pColE1 oriV were 
        indeed functional. Unexpectedly, sample 6769 from this batch lacked a promoter but colonies were still 
        observed. However, the sequencing data for 6769 was unreadable, so we cannot draw any conclusions about its 
        functionality. The observed colonies might very well be due to contamination or high background levels. Upon 
        reviewing our designs, we realized that samples lacking a promoter (like 6769) could theoretically produce 
        functional plasmids if driven by the promoter of the antibiotic resistance gene. We had assembled the 
        resistance gene and oriV in the same orientation, which could allow the antibiotic resistance gene's promoter 
        to inadvertently drive the expression of oriV. To address this issue, we plan to assemble the parts in our next 
        batch such that they run in opposite directions, which is currently ongoing.
      </p>
    </div>
  {{ subheading('Parts') }}
    <div>
      <p>
        All experiments were carried out using the Invitrogen One Shot™ MAX Efficiency™ DH5α-T1R Competent Cells.
      </p>
      <p>
        Table 1. Vectors and synthesized DNA used for the cloning process
      </p>
      <div class="w-full overflow-auto">
        <table class="entrepreneurship">
          <tr>
            <th>Part Number</th>
            <th>Type</th>
            <th>Description</th>
          </tr>
          <tr>
              <td></td>
              <td>PCR Product</td>
              <td>eforRed/eforCP Biobrick K592012 underwent PCR with the bla_part_B_F and bla_part_A_R primers to 
              create a testing vector consisting of the Ampicillin resistance gene (bla) and its promoter, RBS, and 
              terminator. BsaI cut sites and GAGT/CCTT overhangs were added in via PCR.</td>
          </tr>
          <tr>
              <td></td>
              <td>PCR Product</td>
              <td>This consists of the pJUMP27-1A (sfGFP) oriV, and itis a E. coli plasmid origin of replication.<br>
              pJUMP27-1A (sfGFP) underwent PCR with the pJUMP27-1A(sfGFP) 2054-2075 F1 and pJUMP27-1A(sfGFP) 2633-2654 
              R2 primers. BsaI cut sites and AAGG/ACTC overhangs were added in via PCR.
</td>
          </tr>
          <tr>
              <td></td>
              <td></td>
              <td>Wild-type origin of replication from the plasmid pColE1. Batch 1 sequences aligned with this oriV and 
              it functioned as a control to confirm the modularity of oriVs.</td>
          </tr>
          <tr>
              <td></td>
              <td></td>
              <td>AI-generated origin of replication by a tokenizer, named sample 4089. Has a length of 975 bp. It was 
              aligned with the Type B wild-type plasmid ori from E. coli plasmid pT108.</td>
          </tr>
          <tr>
              <td></td>
              <td></td>
              <td>AI-generated origin of replication by a tokenizer, named sample 5276. Has a length of 963 bp. 
              BBa_K5230006 was matched with the Type B wild-type plasmid ori from E. coli plasmid pT108.</td>
          </tr>
          <tr>
              <td></td>
              <td></td>
              <td>AI-generated origin of replication by a tokenizer, named sample 5727. Has a length of 965 bp. 
              BBa_K5230007 was matched with the Type B wild-type plasmid ori from E. coli plasmid pT108.</td>
          </tr>
          <tr>
              <td></td>
              <td></td>
              <td>AI-generated origin of replication by a tokenizer, named sample 9371. Has a length of 688 bp. 
              BBa_K5230008 was matched with the Type B wild-type plasmid ori from E. coli plasmid pT108.</td>
          </tr>
        </table>
      </div>
      <p>
        Table 2. Primers used for cloning process
      </p>
      <div class="w-full overflow-auto">
        <table class="entrepreneurship">
          <tr>
            <th>Part Number</th>
            <th>Type</th>
            <th>Description</th>
          </tr>
          <tr>
              <td></td>
              <td>Primer</td>
              <td>Forward primer for amplicilin resistance backbone eforRed/eforCP K592012. Primer has a length of 36, 
              52.78% GC content, and annealing temperature of 66.2°C. Contains a BsaI cut site and overhang sequence 
              GAGT.</td>
          </tr>
          <tr>
              <td></td>
              <td>Primer</td>
              <td>Reverse primer for the ampicillin resistance backbone eforRed_eforCP K592012. Primer has a length of 
              36 bp, 50% GC content, and annealing temperature of 66.2°C. Contains a BsaI cut site and overhang 
              sequence CCTT.</td>
          </tr>
          <tr>
              <td></td>
              <td>Primer</td>
              <td>Forward primer for positive control OriV pJUMP27-1A(sfGFP). Primer has a length of 36, 55.56% GC 
              content, and annealing temperature of 66.4°C.</td>
          </tr>
          <tr>
              <td></td>
              <td>Primer</td>
              <td>Reverse primer for positive control OriV pJUMP27-1A(sfGFP). Primer has a length of 36, 61.11% GC 
              content, and annealing temperature of 66.4°C.</td>
          </tr>
          <tr>
              <td></td>
              <td>Primer</td>
              <td>Forward primer for the batch one AI-generated OriVs. Primer has a length of 22, 59.09% GC content, 
              and annealing temperature of 67.3°C.</td>
          </tr>
          <tr>
              <td></td>
              <td>Primer</td>
              <td>Reverse primer for the batch one AI-generated OriVs. Primer has a length of 22, 50% GC content, 
              and annealing temperature of 65.8°C.</td>
          </tr>
          <tr>
              <td></td>
              <td>Primer</td>
              <td>Forward primer with a length of 36 base pairs, used during PCR amplification of the wild-type pColE1 
              ori. GC content is 54.5%, and annealing temperature is 66.2C.</td>
          </tr>
          <tr>
              <td></td>
              <td>Primer</td>
              <td>Reverse primer with a length of 36 base pairs, used during PCR amplification of the wild-type pColE1 
              ori. GC content is 50%, and an annealing temperature of 66.2C.</td>
          </tr>
        </table>
      </div>
    </div>
  {{ subheading('Proof of concept') }}
  {{ subheading('Safety') }}
  {{ subheading('Lab Spaces') }}
    <div>
      <p>
        For the safety of our team members, we ensure all experiments are performed in functional, appropriate, 
        isolated and properly organized facilities.
      </p>
      <p>
        All our experiments are performed in a biosafety level 1 facility, with appropriate Personal Protective 
        Equipment open to access. All our laboratory facilities are labelled and documented to the drawer, with proper 
        labeling and segregation mechanisms in place. 
      </p>
    </div>

{% endblock %}
