{% extends "layout/subpage.html" %}

{% block article_content %}
{{ heading("Wet Lab") }}
{{ subheading('Results') }}
<div>
 {{ subsubheading('Controls') }}
  <p>
    We first began by assembling and validating our positive and negative controls. After switching over to the
    eforRed_eforCP vector, we amplified the positive control pJUMP27 insert and verified via agarose gel electrophoresis
    its correct size of 1.6 kb before purifying, transforming, and plating.
  </p>
  <div style="margin: auto; width: 20%; height: auto;">
    {{ img(new_cdn("wetlab/pjump27-gel.webp"), alt="") }}
  </div>
  <p>
 {{ subsubheading('Batch 1') }}
  </p>
  <p>
    Unfortunately, numerous attempts at transforming the first batch of AI-generated sequences did not produce viable
    colonies on the plates. Since our positive control produced colonies, we concluded that this first batch was
    nonfunctional.
  </p>
  <div style="margin: auto; width: 80%; height: auto;">{{ img(new_cdn("wetlab/batch-1-controls.webp"), alt="") }}</div>
  <div style="margin: auto">{{ img(new_cdn("wetlab/batch-1-results.webp"), alt="") }}</div>
  <p>
    The wet lab pipeline for validating various AI-plasmid parts, in particular the <i>oriV</i> and antibiotic resistance genes
    (ARGs), is described in detail in the following sections. Thus far, the wet lab team has synthesized and tested
    novel <i>oriV</i>s generated by dry lab, of which samples 741, 9371, 5276, 5727, and 4089 have been determined as
    functional.
  </p>
  <p>
 {{ subsubheading('Batch 2') }}
  </p>
  <p>
    After completing the required workflow to prepare the inserts, we introduced our second batch of plasmids into E.
    <i>coli</i> for uptake. This batch included eight new AI-generated <i>oriV</i> sequences, six of which produced viable colonies on
    the plates. This result aligned with our expectations, as Batch 2 showed improved alignment with the <i>oriV</i> control
    element RNAII.
  </p>
  <div>{{ img(new_cdn("wetlab/screenshot-2024-10-01-at-16-23-18.webp"), alt="") }}</div>
  <p>
    We also tested the wild-type <i>oriV</i> which the sequences aligned to, pColE, and saw viable colonies, which provides a
    reliable point of comparison for the AI-generated <i>oriV</i>s. The positive control, pJUMP27, also produced viable
    colonies, indicating that <i>oriV</i>s are indeed modular. Upon cloning this batch, we found that sequences with the IDs
    741, 4089, 5276, 5727, 6769, and 9371 produced visible colonies. Upon sequencing, we confirmed that the sequences
    with the ID 741, 4089, 5276, 5727, 9371, and the wild-type pColE1 <i>oriV</i> were indeed functional. Unexpectedly, sample
    6769 from this batch lacked a promoter but colonies were still observed. However, the sequencing data for 6769 was
    unreadable, so we cannot draw any conclusions about its functionality. The observed colonies might very well be due
    to contamination or high background levels. Upon reviewing our designs, we realized that samples lacking a promoter
    (like 6769) could theoretically produce functional plasmids if driven by the promoter of the ARG. We had assembled
    the resistance gene and <i>oriV</i> in the same orientation, which could allow the ARG's promoter to inadvertently drive
    the expression of <i>oriV</i>. To address this issue, we plan to assemble the parts in our next batch such that they run in
    opposite directions, which is currently ongoing.
  </p>
</div>
{{ subheading('Proof of concept') }}
<div>
 {{ subsubheading('Positive Control') }}
  <p>
    The <i>oriV</i> from the pJUMP27-1A (BBa_J428326) vector was cloned into the eforRed/eforCP (BBa_J435099) backbone and
    transformed into E. <i>coli</i> DH5a, which generated visible colonies. This functioned as a positive control and confirmed
    that <i>oriV</i>s are modular.
  </p>
  <div>
  {{ img(new_cdn("wetlab/screenshot-2024-10-01-at-08-39-57.webp"), alt="") }}
  </div>
 {{ subsubheading('5 Functional Sequences') }}
  <p>
    After testing batch 1 of the AI-generated <i>oriV</i>s, we found no functional sequences as no visible colonies were seen.
    Upon testing batch 2 sequences, we found 5 functional sequences, summarized below:
  </p>
  <p>Batch 2 Sequence Testing Result Summary</p>
  <div>
    <table style="width: 100%; margin-bottom: 15px; border: 1px solid #ddd; border-collapse: collapse;">
       <tr style="background-color: #dbcfff;">
            <th style="border: 1px solid #ddd; padding: 8px;">Sequence ID</th>
            <th style="border: 1px solid #ddd; padding: 8px;">306</th>
            <th style="border: 1px solid #ddd; padding: 8px;">741</th>
            <th style="border: 1px solid #ddd; padding: 8px;">1670</th>
            <th style="border: 1px solid #ddd; padding: 8px;">4089</th>
            <th style="border: 1px solid #ddd; padding: 8px;">5276</th>
            <th style="border: 1px solid #ddd; padding: 8px;">5727</th>
            <th style="border: 1px solid #ddd; padding: 8px;">6769</th>
            <th style="border: 1px solid #ddd; padding: 8px;">9371</th>
        </tr>
        <tr>
            <td style="border: 1px solid #ddd; padding: 8px;">Visible colonies?</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #ff7f7f;">No</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #ff7f7f;">No</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
        </tr>
        <tr>
            <td style="border: 1px solid #ddd; padding: 8px;">Verified by whole plasmid sequencing?</td>
            <td style="border: 1px solid #ddd; padding: 8px;">N/A</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
            <td style="border: 1px solid #ddd; padding: 8px;">N/A</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #ff7f7f;">No</td>
            <td style="border: 1px solid #ddd; padding: 8px; background-color: #90ee90;">Yes</td>
        </tr>
    </table>
  </div>
  <p>
    The pColE1 wild-type <i>oriV</i> was also confirmed to be functional. The following is a whole plasmid sequencing result
    showing 100% alignment between the eforRed/eforCP + 9371 sequence in silico assembly and the isolated plasmid:
  </p>
  <div>
  {{ img(new_cdn("wetlab/screenshot-2024-10-01-at-08-41-25.webp"), alt="") }}
  </div>
  <p>
    These results demonstrate our proof of concept, indicating that the AI model generated functional <i>oriV</i>s which allow
    the plasmids to replicate in vivo. Currently, we are testing batch 3 of the AI-generated sequences, which includes
    both <i>oriV</i>s and ARGs.
  </p>
</div>
{% endblock %}
